src/DotNetWorker.Core/Http/HttpRequestDataExtensions.cs

```diff
@@ -37,7 +36,7 @@ public static class HttpRequestDataExtensions
                 return null;
             }
 
-            using (var reader = new StreamReader(request.Body, bufferSize: 1024, detectEncodingFromByteOrderMarks: true, encoding: encoding, leaveOpen: true))
+            using (var reader = new StreamReader(request.Body, bufferSize: 1024, detectEncodingFromByteOrderMarks: true, encoding: encoding ?? Encoding.UTF8, leaveOpen: true))
```

Comment:
For non netfx case, we should not send the default value here, instead we should let the `StreamReader` constructor handle it (as it does today).

For netfx, let's override the value to Encoding.UTF8.

Reply:
I'll mark this as resolved based on what we've discussed. I think we're fine with the behavior here as it would be consistent with the current implementation.